source/module_symmetry/symmetry_rho.cpp

```diff
@@ -11,22 +11,22 @@ Symmetry_rho::~Symmetry_rho()
 
 }
 
-void Symmetry_rho::begin(const int &spin_now) const
+void Symmetry_rho::begin(const int &spin_now, const Charge_Broyden &CHR, const PW_Basis &pw, const Parallel_Grid &Pgrid, const Symmetry &symm) const
 {
 	assert(spin_now < 4);//added by zhengdy-soc
 
 	if(!Symmetry::symm_flag) return;
 #ifdef __MPI
 	// parallel version
-	psymm(CHR.rho[spin_now]);
+	psymm(CHR.rho[spin_now], pw, Pgrid, symm);
 #else
 	// series version.
 	symm.rho_symmetry(CHR.rho[spin_now], pw.ncx, pw.ncy, pw.ncz);
 #endif
 	return;
 }
 
-void Symmetry_rho::psymm(double* rho_part) const
+void Symmetry_rho::psymm(double* rho_part, const PW_Basis &pw, const Parallel_Grid &Pgrid, const Symmetry &symm) const
```

Comment:
Why do we need const for Pgrid and symm? also at line 14

Reply:
@dyzheng asked me to do so to protect the members in these classes.